src/V3Dead.cpp

```diff
@@ -423,10 +423,38 @@ class DeadVisitor final : public VNVisitor {
         }
     }
 
+    void preserveTopIfaces(AstNetlist* rootp) {
+        for (AstNodeModule* modp = rootp->modulesp(); modp && modp->level() <= 2;
+             modp = VN_AS(modp->nextp(), NodeModule)) {
+            for (AstNode* subnodep = modp->stmtsp(); subnodep; subnodep = subnodep->nextp()) {
+                if (AstVar* const varp = VN_CAST(subnodep, Var)) {
+                    if (varp->isIfaceRef()) {
+                        AstIfaceRefDType* ifacerefp = nullptr;
+                        if (VN_IS(varp->subDTypep(), IfaceRefDType)) {
```

Comment:
```suggestion
                        const AstNodeDType* const subtypep = varp->subDTypep()->skipRefp();
                        if (VN_IS(subtypep, IfaceRefDType)) {
```

And similar elsewhere so that typedef's work.

Reply:
Done